src/Miunie.WindowsApp/App.xaml.cs

```diff
@@ -31,8 +35,41 @@ sealed partial class App : Application
         /// </summary>
         public App()
         {
-            this.InitializeComponent();
-            this.Suspending += OnSuspending;
+            InitializeComponent();
+            Suspending += OnSuspending;
+            UnhandledException += App_UnhandledException;
+        }
+
+        private async void App_UnhandledException(object sender, Windows.UI.Xaml.UnhandledExceptionEventArgs e)
+        {
+            e.Handled = true;
+            System.Diagnostics.Debug.WriteLine(e.Exception);
+
+            var logger = SimpleIoc.Default.GetInstance<ILogWriter>();
+            logger.LogError($"{e.Message}");
+
+            var confirmRestart = new ContentDialog
+            {
+                Title = "Opps! Wasn't expecting that...",
+                Content = "Miunie encountered an error that she was unprepared for. Would you like to reset Miunie with a clean restart?",
+                PrimaryButtonText = "Yes, restart Miunie",
+                CloseButtonText = "Try my luck and continue"
+            };
+
+            var confirmResult = await confirmRestart.ShowAsync();
+
+            if (confirmResult == ContentDialogResult.Primary) {
+
+                var restartResult = await CoreApplication.RequestRestartAsync($"Restart Failure: {e.Message}");
+
+                if (restartResult == AppRestartFailureReason.NotInForeground ||
+                    restartResult == AppRestartFailureReason.RestartPending ||
+                    restartResult == AppRestartFailureReason.Other)
```

Comment:
Is there a reason we're not considering the `AppRestartFailureReason.InvalidUser` case?

I haven't looked at what the `RequestRestartAsync` call returns when it goes through, but maybe we could check for that? 🤔 

Or we could hide this boolean expression behind a call at the very least.

Any thoughts on that?